<?xml version="1.0" encoding="UTF-8"?>
<testng-results total="7" passed="5" failed="2" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-01-24T13:37:16Z" name="Surefire suite" finished-at="2024-01-24T13:45:42Z" duration-ms="506380">
    <groups>
    </groups>
    <test started-at="2024-01-24T13:37:16Z" name="Surefire test" finished-at="2024-01-24T13:45:42Z" duration-ms="506380">
      <class name="tests.parts.CounterSale">
        <test-method is-config="true" signature="initialization()[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:07:16Z" name="initialization" finished-at="2024-01-24T19:07:53Z" duration-ms="37699" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- initialization -->
        <test-method signature="goto_countersaleManager_screen()[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:07:53Z" name="goto_countersaleManager_screen" description="Test_001 Login the page and Go to counterSaleManager" finished-at="2024-01-24T19:08:34Z" duration-ms="40291" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- goto_countersaleManager_screen -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:08:34Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:08:34Z" duration-ms="66" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=goto_countersaleManager_screen status=SUCCESS method=CounterSale.goto_countersaleManager_screen()[pri:0, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="verify_all_fields()[pri:1, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:08:34Z" name="verify_all_fields" description="Test_002 verify all the elements in the field" finished-at="2024-01-24T19:08:47Z" duration-ms="12852" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- verify_all_fields -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:08:47Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:08:47Z" duration-ms="82" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verify_all_fields status=SUCCESS method=CounterSale.verify_all_fields()[pri:1, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="add_new_counter_sale()[pri:2, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:08:47Z" name="add_new_counter_sale" description="Test_003 click on add new counter sale" finished-at="2024-01-24T19:10:24Z" duration-ms="96850" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- add_new_counter_sale -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:10:24Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:10:24Z" duration-ms="56" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=add_new_counter_sale status=SUCCESS method=CounterSale.add_new_counter_sale()[pri:2, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="verify_location_popUp()[pri:3, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:10:24Z" name="verify_location_popUp" description="Test_004 verify location pop up" finished-at="2024-01-24T19:11:55Z" duration-ms="90856" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- verify_location_popUp -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:11:55Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:11:55Z" duration-ms="70" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verify_location_popUp status=SUCCESS method=CounterSale.verify_location_popUp()[pri:3, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="verify_counter_sale_page()[pri:4, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:11:55Z" name="verify_counter_sale_page" description="Test_005 verify counter sale page" finished-at="2024-01-24T19:12:27Z" duration-ms="32114" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- verify_counter_sale_page -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:12:27Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:12:27Z" duration-ms="54" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=verify_counter_sale_page status=SUCCESS method=CounterSale.verify_counter_sale_page()[pri:4, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="add_parts_in_counter_sale()[pri:5, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:12:27Z" name="add_parts_in_counter_sale" description="Test_006 add part number and core part in counter sale" finished-at="2024-01-24T19:12:49Z" duration-ms="22184" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at pages.parts.CounterSalePage.addPart(CounterSalePage.java:783)
	at tests.parts.CounterSale.add_parts_in_counter_sale(CounterSale.java:59)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- add_parts_in_counter_sale -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:12:49Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:12:49Z" duration-ms="57" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=add_parts_in_counter_sale status=FAILURE method=CounterSale.add_parts_in_counter_sale()[pri:5, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method signature="check_invoice_and_signout()[pri:6, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:12:49Z" name="check_invoice_and_signout" description="Test_007 check invoice and signout" finished-at="2024-01-24T19:15:42Z" duration-ms="172498" status="FAIL">
          <exception class="org.openqa.selenium.NoSuchElementException">
            <message>
              <![CDATA[no such element: Unable to locate element: {"method":"xpath","selector":"//label[text()='Lost Sale Reason:']/following::a[1]"}
  (Session info: chrome=120.0.6099.225)
For documentation on this error, please visit: https://selenium.dev/exceptions/#no_such_element
Build info: version: '4.4.0', revision: 'e5c75ed026a'
System info: host: 'PC-9', ip: '192.168.29.154', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '11.0.18'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [12a0a3280d148f273f60a14e9442a2c4, findElement {using=xpath, value=//label[text()='Lost Sale Reason:']/following::a[1]}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 120.0.6099.225, chrome: {chromedriverVersion: 120.0.6099.109 (3419140ab66..., userDataDir: C:\Users\pc9-q\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:58294}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:58294/devtoo..., se:cdpVersion: 120.0.6099.225, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 12a0a3280d148f273f60a14e9442a2c4]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.NoSuchElementException: no such element: Unable to locate element: {"method":"xpath","selector":"//label[text()='Lost Sale Reason:']/following::a[1]"}
  (Session info: chrome=120.0.6099.225)
For documentation on this error, please visit: https://selenium.dev/exceptions/#no_such_element
Build info: version: '4.4.0', revision: 'e5c75ed026a'
System info: host: 'PC-9', ip: '192.168.29.154', os.name: 'Windows 11', os.arch: 'amd64', os.version: '10.0', java.version: '11.0.18'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [12a0a3280d148f273f60a14e9442a2c4, findElement {using=xpath, value=//label[text()='Lost Sale Reason:']/following::a[1]}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 120.0.6099.225, chrome: {chromedriverVersion: 120.0.6099.109 (3419140ab66..., userDataDir: C:\Users\pc9-q\AppData\Loca...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:58294}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:58294/devtoo..., se:cdpVersion: 120.0.6099.225, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 12a0a3280d148f273f60a14e9442a2c4
	at jdk.internal.reflect.GeneratedConstructorAccessor24.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:200)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:133)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:53)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:184)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:167)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:142)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:547)
	at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:162)
	at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:60)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:365)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:357)
	at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:70)
	at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:39)
	at com.sun.proxy.$Proxy21.click(Unknown Source)
	at actions.WebElementActions.clickElement(WebElementActions.java:58)
	at pages.parts.CounterSalePage.lostSaleReasonPopup(CounterSalePage.java:919)
	at tests.parts.CounterSale.check_invoice_and_signout(CounterSale.java:71)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.NoSuchElementException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- check_invoice_and_signout -->
        <test-method is-config="true" signature="afterMethodTakeScreenShot(org.testng.ITestResult)[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:15:42Z" name="afterMethodTakeScreenShot" finished-at="2024-01-24T19:15:42Z" duration-ms="48" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=check_invoice_and_signout status=FAILURE method=CounterSale.check_invoice_and_signout()[pri:6, instance:tests.parts.CounterSale@4ba89729] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethodTakeScreenShot -->
        <test-method is-config="true" signature="generateAllure()[pri:0, instance:tests.parts.CounterSale@4ba89729]" started-at="2024-01-24T19:15:42Z" name="generateAllure" finished-at="2024-01-24T19:15:42Z" duration-ms="0" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
	at java.base/java.io.File.<init>(File.java:278)
	at tests.TestDriverActions.generateAllure(TestDriverActions.java:477)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:514)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:215)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:142)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:336)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- generateAllure -->
      </class> <!-- tests.parts.CounterSale -->
    </test> <!-- Surefire test -->
  </suite> <!-- Surefire suite -->
</testng-results>
